src/parser/matches/arg_matches.rs

```diff
@@ -216,6 +216,47 @@ impl ArgMatches {
         MatchesError::unwrap(id, self.try_get_many(id))
     }
 
+    /// Iterate over groups of values of a specific option.
```

Comment:
Could we have a PR per one of these commits?  Each one does enough on its own to have its own conversation

Reply:
Limited this to just the first commit. The second one depends on this though, so I'll wait to make a PR on that until this has settled.